Examples/nn_frameworks/nnx.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import netket as nk
+
+import jax
+import jax.numpy as jnp
+from flax import nnx
+import optax
+
+import numpy as np
+
+
+class RBM(nnx.Module):
+    def __init__(
+        self, N, alpha, rngs: nnx.Rngs, visible_bias: bool = True, param_dtype=complex
+    ):
+        self.linear = nnx.Linear(N, alpha * N, param_dtype=param_dtype, rngs=rngs)
+        if visible_bias:
+            self.visible_bias = nnx.Param(
+                jax.random.uniform(rngs.params(), (N,), dtype=param_dtype)
+            )
+        else:
+            self.visible_bias = False
+
+    def __call__(self, x_in):
+        y = nk.nn.log_cosh(self.linear(x_in))
+        y = jnp.sum(y, axis=-1)
+        if self.visible_bias is not None:
+            # print(y)
+            # print(self.visible_bias)
+            # print("type", type(self.visible_bias))
+            y = y + jnp.dot(x_in, self.visible_bias.raw_value)
+
+        return y
+
+    def another_fun(self, x_in):
```

Comment:
?

Reply:
It was to show that one can also define other functions (like what you need to do autoregressive networks).
It's not very explicative like that tho, indeed.